2024/calcite-react/demo/src/App.tsx

```diff
@@ -30,16 +28,12 @@ function App({ webmap, title, panelHeading }: AppProps) {
       {/* Header slot */}
       <Header title={title} />
       {/* Panel Start slot */}
-      <Panel view={view} panelHeading={panelHeading} />
+      <Panel map={map} panelHeading={panelHeading} />
       {/* Default content slot */}
       <ArcgisMap
         itemId={webmap}
-        // Step 10: Listen for on view ready
-        onViewReady={(e) => {
-          const { view } = e.detail;
-          // Store view in state, and pass that into panel
-          setView(view);
-        }}
+        // Step 10: Use onArcgisViewReadyChange to store map in state to pass into panel
+        onArcgisViewReadyChange={(e) => setMap(e.target)}
```

Comment:
this will only set the map state after the map is ready
but the Features widget in the Panel would already try (and fail) to initialize itself by that point

would this work instead?

```suggestion
        ref={setMap}
```

Reply:
btw, the reason the event is called `onArcgisViewReadyChange` rather than `onArcgisViewReady` is because map view could become un-ready (I don't remember the exact condition, but might be when you add a new layer dynamically)
not an issue in this demo app, but just a FYI in case you get this question at dev summit

Reply:
Nope, that doesn't work